### PR TITLE
WIP audio-only: display 1 screen-share while in audio-only mode

### PR DESCRIPTION
**Pull request by saghul**
This in turn becomes a "low bandwidth mode".

### CODE REVIEW

File: react/features/mobile/background/actions.js [hunk 1]
```diff
@@ -40,13 +40,21 @@ export function _setAppStateListener(listener: ?Function) {
...
+            dispatch(setLastN(0));
+        } else {
+            const state = getState();
+            const { audioOnly } = state['features/base/conference'];
+            const screenShares = state['features/video-layout'].screenShares || [];
+            const latestScreenshare = screenShares[screenShares.length - 1];
 
-        audioOnly || dispatch(setLastN(muted ? 0 : undefined));
+            if (audioOnly) {
+                dispatch(setLastN(latestScreenshare ? 1 : 0));
```
Comment (zbettenbuk) [inline react/features/mobile/background/actions.js]: Why latestScreenshare? Shouldn't it be enough to just use screenShares.length?

File: react/features/video-layout/subscriber.js [hunk 2]
```diff
@@ -111,7 +115,13 @@ function _updateAutoPinnedParticipant({ dispatch, getState }) {
 
     if (latestScreenshareParticipantId) {
         dispatch(pinParticipant(latestScreenshareParticipantId));
-    } else if (getPinnedParticipant(state['features/base/participants'])) {
+
+        // If we are in audio-only mode, set last N to 1.
+        appState === 'active' && audioOnly && dispatch(setLastN(1));
```
Comment (zbettenbuk) [inline react/features/video-layout/subscriber.js]: Won't this leave lastN to 1 even if a non-screenshare participant is pinned?

File: react/features/video-layout/subscriber.js [hunk 3]
```diff
@@ -101,6 +102,9 @@ StateListenerRegistry.register(
 function _updateAutoPinnedParticipant({ dispatch, getState }) {
     const state = getState();
     const screenShares = state['features/video-layout'].screenShares;
+    const pinnedParticipant = getPinnedParticipant(state['features/base/participants']);
+    const { audioOnly } = state['features/base/conference'];
+    const { appState } = state['features/background'];
 
     if (!screenShares) {
```
Comment (zbettenbuk) [inline react/features/video-layout/subscriber.js]: I know it's not your change, but can we return earlier with this condition?

File: react/features/mobile/background/actions.js (hunk 1 above)
Comment (saghul) [reply react/features/mobile/background/actions.js]: Yes indeed

File: react/features/video-layout/subscriber.js (hunk 2 above)
Comment (saghul) [reply react/features/video-layout/subscriber.js]: Yes, but note that 2 lines above we pin the screen-sharing participant. The plan is to also "lock" the pinning capability...

**Pull request closed without merging**